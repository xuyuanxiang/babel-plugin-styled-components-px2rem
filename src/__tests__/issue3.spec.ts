import { transformFileSync } from '@babel/core';
import { join } from 'path';
import plugin from '../';

describe('issue#3', () => {
  it('should support multi-level code blocks', () => {
    const result = transformFileSync(join(__dirname, 'issue3.txt'), {
      plugins: [plugin],
      presets: ['@babel/preset-react'],
    });
    if (result && result.code) {
      expect(result.code).toMatchSnapshot();
    } else {
      throw new Error('Should transform code');
    }
  });
});
