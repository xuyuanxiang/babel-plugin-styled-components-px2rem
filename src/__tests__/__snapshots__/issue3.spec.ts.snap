// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`issue#3 should support multi-level code blocks 1`] = `
"const layout1 = styled.div\`
  & {
    position: relative;
  }

  & {
    footer {
      height: 0.8rem;
      background: green;
    }

    header {
      height: 0.45rem;
      background: yellow;
    }
  }
\`;
const layout2 = styled.div\`
  & {
    position: relative;
  }

  & footer {
    height: 0.8rem;
    background: green;
  }

  & header {
    height: 0.45rem;
    background: yellow;
  }
\`;
const layout3 = styled.div\`
  & footer {
    height: 0.8rem;
    background: green;
  }

  & header {
    height: 0.45rem;
    background: yellow;
  }
\`;"
`;
